### PR TITLE
MOTECH-1966: Generated PDFs can now be easily printed (0.27.X)

### PR DESCRIPTION
**Pull request by ngraczewski**

### CODE REVIEW

**Comment by slaid3r** [comment]
test this

**Comment by slaid3r** [comment]
test this please

**Comment by pgesek** [comment]
test this please

File: platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/csv/writer/PdfTableWriter.java [hunk 1]
```diff
@@ -108,15 +194,35 @@ private float calculateTotalTableWidth(float[] widths) {
...
-            columnsWidths.put(header, width > MAX_COLUMN_WIDTH ? MAX_COLUMN_WIDTH : width);
+        for (int column = firstColumn; column < lastColumn; column++) {
+            totalColumnsSize += relativeWidths[column];
+        }
+
+        return totalColumnsSize;
+    }
+
+    private List<Integer> calculateLastColumnsForPages(float[] relativeWidths) {
+        List<Integer> lastColumnsForPages = new LinkedList<>();
```
Comment (sebbrudzinski) [inline platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/csv/writer/PdfTableWriter.java]: Why LinkedList? I can see you use get by index on the list, which for LinkedList has complexity O(n), while for ArrayList this is obviously O(1). Unless you plan to access/delete elements with iterators use ArrayLists

File: platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/csv/writer/PdfTableWriter.java [hunk 2]
```diff
@@ -28,51 +29,54 @@
...
+    private static final float PAGE_HEIGHT = PageSize.A4.getWidth() - 2 * MARGIN;
+    private static final float PAGE_WIDTH = PageSize.A4.getHeight() - 2 * MARGIN;
 
     private final PdfWriter pdfWriter;
     private final Document pdfDocument;
+    private final PdfContentByte pdfCanvas;
     private PdfPTable dataTable;
     private Map<String, Float> columnsWidths;
+    private int rows = 0;
+    private float tableContentX;
```
Comment (sebbrudzinski) [inline platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/csv/writer/PdfTableWriter.java]: Can this be more descriptive? I assujme X is some size?

**Comment by sebbrudzinski** [comment]
@pgesek OK to merge

**Pull request merged**